libs/calladapt/calladapt.go

```diff
@@ -0,0 +1,157 @@
+package calladapt
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// TypeOf returns reflect.Type for type parameter T, analogous to
+// reflect.TypeOf((*T)(nil)).Elem().
+func TypeOf[T any]() reflect.Type {
+	var t *T
+	return reflect.TypeOf(t).Elem()
+}
+
+// BoundCaller encapsulates a bound method and metadata about its signature.
+// It can invoke the underlying function and returns all non-error outputs and
+// the error (if the method returns one as the last return value).
+type BoundCaller struct {
```

Comment:
It would be helpful to have docstrings for the fields in BoundCaller.

Reply:
Sure e5c194126c46d359a0748cbc9043818c91441d02